Use nightly compiler, support building C code

arturkow2000 opened the pull request.

0xDen reviewed (approved):
LGTM

macpijan commented:
This repo is archived. The rust SDK was moved here: https://github.com/fobnail/fobnail-sdk

arturkow2000 committed:
Dockerfile: use latest Rust nightly compiler (from 2022-01-13)

arturkow2000 committed:
Add packages needed to compile C code.

Fobnail firmware now depends on Trussed, which depends on C
implementation of littlefs2.

0xDen commented on Dockerfile:
```diff
@@ -22,15 +26,20 @@ RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o /tmp/rustup-ini
         --default-toolchain nightly-2022-01-13 \
-        -t thumbv7em-none-eabihf && \
+        -t thumbv7em-none-eabihf \
+	-c llvm-tools-preview \
```
If `-c llvm-tools-prevew` is the option for `/tmp/rustup-init` then could be aligned with others?

arturkow2000 replied:
Fixed.

krystian-hebel replied:
Not fixed, this is not in the same column as `-t ...` above.

arturkow2000 committed:
Dockerfile: fix indentation (replace tabs with spaces)

krystian-hebel committed:
Dockerfile: fix indentation level

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel reviewed (approved):
Docker image is able to build and run both nRF and PC code with this change.
